### PR TITLE
add environments to skylight

### LINKED ISSUE #8103: Add stable and unstable environments to skylight for monitoring
**Please describe the desired behavior.**
Currently skylight only monitors the production server, adding the stable and unstable environments to skylight will allow us to push our changes on these servers and monitor performance metrics and db queries.

### PR DESCRIPTION
**Pull request by Tlazypanda**
Fixes #8103

Added stable and unstable environments to skylight.

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

Thanks!

### CODE REVIEW

**Comment by Tlazypanda** [comment]
Hey @jywarren @cesswairimu can you kindly review? Thanks :v:

**Comment by Tlazypanda** [comment]
Hey @jywarren I think the tests are passing when you check it on the travis pr page but its status on github is not changing :sweat_smile:

**Comment by jywarren** [comment]
Hmmmmmm something is fishy here.

**Comment by jywarren** [comment]
These at least don't seem to be completed: https://github.com/publiclab/plots2/pull/8104/checks?check_run_id=825545745

**Comment by jywarren** [comment]
The ones that aren't passing here are from 6 days ago. Would you mind trying to rebase this to see if we get a clean run? I wonder if it's related to the switch to Travis-ci.com from Travis-ci.org?

https://github.com/publiclab/plots2/pull/8104/checks?check_run_id=825545745

Thank you and sorry for the trouble!

**Comment by Tlazypanda** [comment]
Hey @jywarren actually I opened the jobs on the link that you sent which show as yellow now but on opening them it is green :sweat_smile: is this from before test?

**Comment by Tlazypanda** [comment]
Hey @jywarren just adding to the slow travis test point that you mentioned, the tests in this run are still queued and haven't even started yet https://travis-ci.com/github/publiclab/plots2/jobs/358427078 :sweat_smile:

**Comment by jywarren** [comment]
Hmm. I wonder if we're hitting a rate limit. I have restarted a TON of travis tests today.

**Comment by jywarren** [comment]
This looks great! Thank you!!! cc @icarito bc we would have split config files. Just checking that's OK?

**Comment by icarito** [comment]
Hi!
Sorry for the delay in reviewing.
I see a lot of duplication here, could this rather be accomplished by environment variables rather than having one config per environment?
The idea currently is that staging runs in `production` mode so that it is as close as possible to actual production.
We currently have one config for each environment in `containers/` but I would also like to refactor those into one with env variables.

**Comment by jywarren** [comment]
Hi @tlazypanda do you have any insight on this q from @icarito? Thank you!

File: config/environments/staging.rb [hunk 1]
```diff
@@ -0,0 +1,89 @@
+Plots2::Application.configure do
```
Comment (jywarren) [inline config/environments/staging.rb:1]: Do these extend the existing configs, or replace them? Thanks!

**Comment by Tlazypanda** [comment]
Hey @jywarren since this is a new area for me I am not completely sure how to implement on ENV variables :sweat_smile: I also did some digging but couldn't find much around it catered to rails environments :sweat_smile: As for the extending/replacing thing I think it's replacing on it since that is how it is in the other development/test environments as well and I have copied the entire production config here and just changed the hosts :sweat_smile:

**Comment by jywarren** [comment]
Hi, well, if you are pretty sure this will only affect the skylight run and environment, I think this is OK! Let's merge it - if we find something goes wrong, we can revert back, it's not such a big thing. Thanks and I appreciate your patience as we're pretty careful with this!

**Comment by jywarren** [comment]
🎉

**Comment by jywarren** [comment]
Just noting docs for this here:

https://www.skylight.io/support/environments

If it doesn't work, we could revert and try the env variable approach listed there!

https://www.skylight.io/support/environments#for-rails-using-environment-variables

@Tlazypanda i invited you to our Skylight account! I don't see the new env yet, but, hopefully soon!

**Comment by Tlazypanda** [comment]
Hey @jywarren I was just checking this out could it be because it's not deployed yet (I am not sure about this but I cant see the deploy rocket line in the latest time slots) :sweat_smile: 

https://www.skylight.io/support/skylight-guides#deploy-tracking
https://www.skylight.io/support/troubleshooting#did-you-deploy-your-app-since-setting-up-skylight

I am not completely sure if that's how it works with git repo connected to skylight but since it is studying from publiclab.org and not stable.publiclab.org could it be a reason? :sweat_smile:  (just adding here I have no idea about this so I am really sorry if I totally misread this :crying_cat_face:)

**Comment by Tlazypanda** [comment]
Hey @jywarren I am not sure why the environment is not showing :sweat_smile: can we try the second approach?

**Comment by icarito** [comment]
I think https://www.skylight.io/support/environments#3-optional-specify-an-environment-name seems to indicate that the SKYLIGHT_ENV needs to be set if we don't change RAILS_ENV (and we don't, staging mimics production so it has same RAILS_ENV). 
So I've set SKYLIGHT_ENV for unstable branch in staging, and also made relevant changes to docker-compose-unstable.yml file. If it works we can extend the solution and set more environment variables.

**Comment by jywarren** [comment]
@Tlazypanda did that end up working out?

**Comment by Tlazypanda** [comment]
Hey @jywarren I think what @icarito has done works :tada: since I can see unstable as an option in the skylight dropdown :100: 
Thanks @icarito!!

**Pull request merged**